lib/Frontend/CompileJobCacheKey.cpp

```diff
@@ -0,0 +1,81 @@
+//===--- CompileJobCacheKey.cpp - compile cache key methods ---------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains utility methods for creating compile job cache keys.
+//
+//===----------------------------------------------------------------------===//
+
+#include <swift/Frontend/CompileJobCacheKey.h>
+#include <swift/Basic/Version.h>
+#include <llvm/ADT/SmallString.h>
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/CAS/HierarchicalTreeBuilder.h"
+#include "llvm/CAS/ObjectStore.h"
+
+using namespace swift;
+
+// TODO: Rewrite this into CASNodeSchema.
+llvm::Expected<llvm::cas::ObjectRef> swift::createCompileJobBaseCacheKey(
```

Comment:
How are input file contents accounted for here?

Reply:
It is expected to come from `-cas-fs` argument. Currently, swift-frontend will produce the cache key even when `-cas-fs` is not used. This is risky since file content are in the cache key and the cache entry can't really be used but on the other hand, it makes writing unit test easier in swift repo since compute `cas-fs` for a swift invocation is not easy.

Reply:
Makes sense. How about adding a warning if there is no cas-fs?  Would be good to explain this in a comment here too.  I'm fine with tests cheating here, just don't want to accidentally use this in production 😸

Reply:
True. Let me actually just add a hidden flag like `-allow-unstable-cache-key-for-test` which we can use in test, otherwise, it will just error out if caching is used without `-cas-fs`